### ISSUE TITLE
"Whitespace around operators" check reports the same warning twice

### ISSUE THREAD

**Issue by barrywhart** [alanmcruickshank/sqlfluff#74]
Query:
```
SELECT
    a/b AS click_rate
FROM
    t
```

Output:
```
$ sqlfluff lint test.sql
== [test.sql] FAIL
L:   2 | P:   6 | L006 | Operators should be surrounded by a single whitespace.
L:   2 | P:   7 | L006 | Operators should be surrounded by a single whitespace.
```

Note the warning is correct, but it is reported twice -- apparently once on the operator and once on the second operand.

**Comment by alanmcruickshank** [comment]
In this case, it's reporting a missing whitespace *before* the operator *and* a missing whitespace *after* the operator. At the moment that's the intended functionality, but do you think that's confusing for the user?

**Comment by barrywhart** [comment]
That explanation makes sense. I think it would be reasonable to emit a single message, or perhaps word the two messages differently, e.g.:
* `"Operators should be preceded by a space."`
* `"Operators should be followed by a space."`

**Comment by barrywhart** [comment]
I saw the fix PR for this -- thanks! Maybe someday I can help on some of the parse or autoformatting bits.

This is going to be an incredibly useful package! I'm not aware of any good SQL linters, and providing autoformatting/fixing in the same package is 💯 🎉 .

**Comment by alanmcruickshank** [comment]
@barrywhart thanks!! What I'd love to do is start getting this into live deployment pipelines for sql code. I'm currently prioritising fixes which will get us there for the projects that I look after. If you work on any projects which this would be valuable for I'd love to start an "in the wild" list to start people seeing how it's being used properly.

**Comment by barrywhart** [comment]
Yes! I added it to a production project this week (that's where I found the issues), and it now runs for every PR. It doesn't cause builds to fail, but the output is visible in the Jenkins build log. I also added it to a second project that is just getting started -- I expect it to really ramp up after Christmas.

I want to add some tooling to run it and filter the output to the new or modified parts of the code, similar to the diff-cover package -- are you familiar with it?

At that point, it may make sense to start having it fail the builds.

**Comment by alanmcruickshank** [comment]
@barrywhart - I like the idea of being able to use the library as part of tooling. I'm not familiar with diff-cover, but I could see two ways of implementing that:
1. We build a load of features into sqlfluff for it to filter it's output based on some input diff file.
2. We expose sqlfluff as a library you can sensibly `import` from, and then you can take the output of sqlfluff and do the filtering downstream in your own tooling.

I'm open to either suggestion - can you open a new issue for it because this one is closed? It would be awesome if you could summarise some of this conversation into that card so it's easy to find the relevant bits.

**Comment by barrywhart** [comment]
Will do! I'll try and research a bit about diff-cover as well.
